api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -67,24 +70,52 @@ default UpdateSchema addColumn(String name, Type type) {
   }
 
   /**
-   * Add a new top-level column.
+   * Add a new optional top-level column.
    *
    * <p>Because "." may be interpreted as a column path separator or may be used in field names, it
    * is not allowed in names passed to this method. To add to nested structures or to add fields
    * with names that contain ".", use {@link #addColumn(String, String, Type)}.
    *
    * <p>If type is a nested type, its field IDs are reassigned when added to the existing schema.
    *
+   * <p>The added column will be optional with a null default value.
+   *
    * @param name name for the new column
    * @param type type for the new column
    * @param doc documentation string for the new column
    * @return this for method chaining
    * @throws IllegalArgumentException If name contains "."
    */
-  UpdateSchema addColumn(String name, Type type, String doc);
+  default UpdateSchema addColumn(String name, Type type, String doc) {
```

Comment:
Have we ever considered supporting `NestedField` in methods that add columns and relying on its builder?